Automattic/woocommerce-payments
# Remove redundant compatibility util class and refs

Fixes #5291

#### Changes proposed in this Pull Request

Some time ago, there was a utility class - `WC_Payment_Woo_Compat_Utils` [was introduced](https://github.com/Automattic/woocommerce-payments/commit/bc7e5527c38582ba7f727fb7341874fdc962d469#diff-a6cc377d2eea7044c565e504ec680653441aad675b64a1900d23027846f33f8d),  to ensure older versions of WooCommerce ( Prior to migration of WC Admin classes to core ), would render WC Pay Admin notes correctly on the settings page. 

The required classes of WC Admin, including that of the Notes component, were subsequently [migrated to WooCommerce Core](https://github.com/woocommerce/woocommerce/commits/3dd60322d5490d7d86ad33e7d732f276df73a12d/plugins/woocommerce/src/Admin/Notes/Note.php) and the util class - `WC_Payment_Woo_Compat_Utils` is now redundant. 

The PR removes this util class, and changes all `Note` and `Notes` component references to that of Woo Core class.

#### Testing instructions

* Test on current WooCommerce version - Browse to `Payments > Settings` within WP Admin, and check if the notes appear correctly. 

![image](https://user-images.githubusercontent.com/4162931/209345898-b7eaf4a6-af4f-478f-b66a-98ecfb22772a.png)

* Check the console for any errors too
* Try closing a note and see if there are any errors, and reload to see if it remains removed
* Test on minimum supported WooCommerce version ( Currently Version 7.0.0)
* Repeat the first three steps of testing with this older version and make sure it works without any errors
* Run unit tests too and make sure it clears. 

#### Advice needed

I see that in one of the files, a ` \`  [was used preceding](https://github.com/Automattic/woocommerce-payments/pull/5331/files#diff-f8ba4a7821e5ea4491b18fd1db1a1fa2a7784545e7f0f313a8cad84622e7524eL43) the class name indicating global scope. I am not 100% sure of the implication of this here. Is it needed to be added on the changed class too? ( I think no, but thought of asking to be safe ) 

-------------------


**Post merge**
